### PR TITLE
Upgrading to NDK R10e and gcc 4.9

### PR DESCRIPTION
**Pull request by kneth**
In order to integrate the object store and upcoming storage engine changes, we need to upgrade to gcc 4.9.

This PR cannot be merged until the storage engine releases a version using NDK R10e and gcc 4.9 (notice "waiting-core" label).

 architecture | old size      | new size      
 :----------------|----------------:|----------------:
 arm64          | 1873880    | 1882072       
 arm              | 1115580    | 1160644      
 armv7          | 1115588    | 1144268      
 mips            | 2304576    | 2368544      
 x86              | 1779172    | 1824236      
 x86_64        | 2038024    | 2029832      

TODO:
* [x] Change `ext.coreVersion` and `ext.coreSha256Hash`.
* [x] Update `README.md`

@realm/java

### CODE REVIEW

**Comment by cmelchior** [comment]
Since the compile flags have changed. What does this mean for the final compiled size and performance?

**Comment by kneth** [comment]
They didn't really change. They are set two different places, and now they are consistent. I believe that previously the full set of options were the same. Actually, not sure where the right place to set them is these days.

**Comment by kneth** [comment]
@realm/java I have updated the description with a table to compare sizes between this build and our latest release.

File: realm/realm-jni/project.mk [hunk 1]
```diff
@@ -52,7 +52,7 @@ ifeq ($(REALM_ANDROID),)
     endif
   endif
 else
-  PROJECT_CFLAGS += -fvisibility=hidden -DANDROID
+  PROJECT_CFLAGS += -fvisibility=hidden -DANDROID -ffunction-sections -fdata-sections -flto
```
Comment (emanuelez) [inline realm/realm-jni/project.mk]: Is this file used? AFAIR We use a normal makefile and not an android one.

**Comment by cmelchior** [comment]
The README.md also contains a reference how to install `r10d` that needs to be updated as well.

File: realm/realm-jni/project.mk (hunk 1 above)
Comment (kneth) [inline realm/realm-jni/project.mk]: Can we remove it?

**Comment by kneth** [comment]
I have updated to latest core. I'm not sure if I should target `master` or a release branch.

**Comment by kneth** [comment]
retest this please

File: realm/realm-jni/project.mk (hunk 1 above)
Comment (beeender) [inline realm/realm-jni/project.mk]: This file is included by the `generic.mk` but `PROJECT_CFLAGS` doesn't seem to be used by `realm-jni` . Is this change from core's change?

**Comment by emanuelez** [comment]
retest this please

File: realm/realm-jni/project.mk (hunk 1 above)
Comment (kneth) [inline realm/realm-jni/project.mk]: I added them for consistency. I hope we can move to a "pure" Gradle/Android Studio setup soon.

**Comment by emanuelez** [comment]
retest this please

**Comment by kneth** [comment]
I close this PR as #1976 targets our latest release.

**Pull request closed without merging**